lab_2_tokenize_by_bpe/main.py

```diff
@@ -69,6 +154,28 @@ def get_vocabulary(
     :param unknown_token: a token to signify an unknown token
     :return: dictionary in the form of <token: identifier>
     """
+    if not (
+        isinstance(word_frequencies, dict) and
+        isinstance(unknown_token, str)
+    ):
+        return None
+
+    identificators = {}
+    all_tokens = [unknown_token]
+
+    for tuple_word in word_frequencies:
+        for word in tuple_word:
+            all_tokens.append(word)
+            prepared_word = prepare_word(word, None, None)
+            if not prepared_word:
+                return None
+            all_tokens += list(prepared_word)
+    sorted_token_set = sorted(set(all_tokens), key=lambda x: (-len(x), x))
```

Comment:
Set is an unordered structure, let's not sort it. Consider casting to list or something.

Reply:
fixed